lib/process-release.js

```diff
@@ -0,0 +1,87 @@
+var semver = require('semver')
+var url = require('url')
+var path = require('path')
+
+var bitsre = /(x32|x64)/
+
+function processRelease (argv, gyp, defaultVersion, defaultRelease) {
```

Comment:
we don't look up anything on `process` here, those are passed in by the caller, this is mainly about keeping this self-contained and to make it possible to test it against different versions of those properties